app/uk/gov/hmrc/economiccrimelevyenrolment/forms/EclReferenceFormProvider.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.economiccrimelevyenrolment.forms
+
+import play.api.data.Form
+import uk.gov.hmrc.economiccrimelevyenrolment.forms.mappings.Mappings
+
+import javax.inject.Inject
+
+class EclReferenceFormProvider @Inject() extends Mappings {
+
+  def apply(): Form[String] =
```

Comment:
We can apply regex validation to this, as we have the regex for an ECL reference, would just need to think about the error message